src/internal/schema.ts

```diff
@@ -68,27 +70,29 @@ export function _buildSchema<T, U extends AnyParamConstructor<T>>(
 
   sch.loadClass(cl);
 
-  // Hooks
-  {
-    /** Get Metadata for PreHooks */
-    const preHooks: IHooksArray[] = Reflect.getMetadata(DecoratorKeys.HooksPre, cl);
-    if (Array.isArray(preHooks)) {
-      preHooks.forEach((obj) => sch.pre(obj.method, obj.func as EmptyVoidFn));
+  if (isFinalSchema) {
+    // Hooks
+    {
+      /** Get Metadata for PreHooks */
+      const preHooks: IHooksArray[] = Reflect.getMetadata(DecoratorKeys.HooksPre, cl);
+      if (Array.isArray(preHooks)) {
+        preHooks.forEach((obj) => sch.pre(obj.method, obj.func as EmptyVoidFn));
+      }
+
+      /** Get Metadata for PreHooks */
+      const postHooks: IHooksArray[] = Reflect.getMetadata(DecoratorKeys.HooksPost, cl);
+      if (Array.isArray(postHooks)) {
+        postHooks.forEach((obj) => sch.post(obj.method, obj.func));
+      }
     }
 
-    /** Get Metadata for PreHooks */
-    const postHooks: IHooksArray[] = Reflect.getMetadata(DecoratorKeys.HooksPost, cl);
-    if (Array.isArray(postHooks)) {
-      postHooks.forEach((obj) => sch.post(obj.method, obj.func));
-    }
-  }
-
-  /** Get Metadata for indices */
-  const plugins: IPluginsArray<any>[] = Reflect.getMetadata(DecoratorKeys.Plugins, cl);
-  if (Array.isArray(plugins)) {
-    for (const plugin of plugins) {
-      logger.debug('Applying Plugin:', plugin);
-      sch.plugin(plugin.mongoosePlugin, plugin.options);
+    /** Get Metadata for indices */
+    const plugins: IPluginsArray<any>[] = Reflect.getMetadata(DecoratorKeys.Plugins, cl);
+    if (Array.isArray(plugins)) {
+      for (const plugin of plugins) {
+        logger.debug('Applying Plugin:', plugin);
+        sch.plugin(plugin.mongoosePlugin, plugin.options);
+      }
     }
   }
```

Comment:
Please add `virtuals`, `indices`, `queryMethods` and the `sch.method` call to the `isFinalSchema` "if"